lib/membrane_http_adaptive_stream/manifest/track.ex

```diff
@@ -185,25 +229,40 @@ defmodule Membrane.HTTPAdaptiveStream.Manifest.Track do
   def supports_partial_segments?(%__MODULE__{target_partial_segment_duration: duration}),
     do: duration != nil
 
-  @doc """
-  Add a segment of given duration to the track.
-  It is recommended not to pass discontinuity attribute manually but use `discontinue/1` function instead.
-  """
+  @spec is_persisted?(t()) :: boolean()
+  def is_persisted?(%{persist?: persist?}), do: persist?
+
+  @spec new_segment(
+          t(),
+          segment_opt_t(),
+          list(Manifest.SegmentAttribute.t())
+        ) :: {Changeset.t(), t()}
+  def new_segment(track, opts, attributes \\ [])
+
+  def new_segment(
+        %__MODULE__{target_partial_segment_duration: nil} = track,
+        opts,
+        attributes
+      ),
+      do: add_segment(track, opts, attributes)
+
+  def new_segment(track, opts, attributes),
+    do: new_partial_segment(track, opts, attributes)
```

Comment:
I'd put the body of `new_partial_segment` here instead of calling it

Reply:
Great idea!